src/ralph/models/edx/problem_interaction/statements.py

```diff
@@ -0,0 +1,303 @@
+    page: Literal["x_module"]
+
+
+class EdxProblemHintFeedbackDisplayed(BaseServerModel):
+    """Represents the `edx.problem.hint.feedback_displayed` server event.
+
+    This event is triggered when a user receives a hint after answering a problem.
+
+    Attributes:
+        event (dict): See EdxProblemHintFeedbackDisplayedEventField.
+        event_type (str): Consists of the value `edx.problem.hint.feedback_displayed`.
+        page (str): Consists of the value `x_module`.
+    """
+
+    __selector__ = selector(
+        event_source="server", event_type="edx.problem.hint.feedback_displayed"
+    )
+
+    event: EdxProblemHintFeedbackDisplayedEventField
+    event_type: Literal["edx.problem.hint.feedback_displayed"]
+    page: Literal["x_module"]
+
+
+class UIProblemCheck(BaseBrowserModel):
+    """Represents the `problem_check` browser event.
+
+    The browser emits this event when a user checks a problem.
+
+    Attributes:
+        event (str): Consists of values of problem being checked, styled as `GET` parameters.
+        event_type (str): Consists of the value `problem_check`.
+        name (str): Consists of the value `problem_check`.
+    """
+
+    __selector__ = selector(event_source="browser", event_type="problem_check")
+
+    event: str
+    event_type: Literal["problem_check"]
+    name: Literal["problem_check"]
+
+
+class ProblemCheck(BaseServerModel):
+    """Represents the `problem_check` server event.
+
+    This event is triggered when a user checks a problem.
+
+    Attributes:
+        event (dict): See ProblemCheckEventField.
+        event_type (str): Consists of the value `problem_check`.
+        page (str): Consists of the value `x_module`.
+    """
+
+    __selector__ = selector(event_source="server", event_type="problem_check")
+
+    event: ProblemCheckEventField
+    event_type: Literal["problem_check"]
+    page: Literal["x_module"]
+
+
+class ProblemCheckFail(BaseServerModel):
+    """Represents the `problem_check_fail` server event.
+
+    This event is triggered when a user checks a problem and a failure prevents the problem
+    from being checked successfully.
+
+    Attributes:
+        event (dict): See ProblemCheckFailEventField.
+        event_type (str): Consists of the value `problem_check_fail`.
+        page (str): Consists of the value `x_module`.
+    """
+
+    __selector__ = selector(event_source="server", event_type="problem_check_fail")
+
+    event: ProblemCheckFailEventField
+    event_type: Literal["problem_check_fail"]
+    page: Literal["x_module"]
+
+
+class UIProblemGraded(BaseBrowserModel):
+    """Represents the `problem_graded` browser event.
+
+    Attributes:
+        event (list): See ProblemGradedEventField.
+        event_type (str): Consists of the value `problem_graded`.
+        name (str): Consists of the value `problem_graded`.
+    """
+
+    __selector__ = selector(event_source="browser", event_type="problem_graded")
+
+    event: List[UIProblemGradedEventField]
```

Comment:
```suggestion
    event: List[Union[str, Literal[None], None]]
```
The event field holds a list of strings (generally of length 2 but not 100% sure)
I tried to document it [here](https://github.com/SergioSim/ralph/blob/docs/docs/browserLogs.md#event_type-problem_graded).